partiql-test-data/fail/parser/primitives/call.ion

```diff
@@ -0,0 +1,174 @@
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING sql92 syntax missing FROM or comma",
+            statement: "SELECT SUBSTRING('str' 1) FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING sql92 syntax without length and missing right paren",
+            statement: "SELECT SUBSTRING('str' FROM 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING sql92 syntax with length missing right paren",
+            statement: "SELECT SUBSTRING('str' FROM 1 FOR 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        }
+    ],
+    partiql::[
+        parse::{
+            name: "SUBSTRING without length missing right paren",
+            statement: "SELECT SUBSTRING('str', 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        },
+        parse::{
+            name: "SUBSTRING missing right paren",
+            statement: "SELECT SUBSTRING('str', 1, 1 FROM foo",
+            assert: {
+                result: ParseError
+            },
+        }
+    ]
+]
+
+trim::[
+    parse::{
+        name: "TRIM no left paren",
+        statement: "TRIM ' ')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM too many arguments",
+        statement: "TRIM(BOTH ' ' FROM 'test' 2)",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM with BOTH missing FROM",
+        statement: "TRIM(BOTH 'test')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM and remove without FROM",
+        statement: "trim(both '' 'test')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM without string",
+        statement: "TRIM(FROM)",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM no args",
+        statement: "TRIM()",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM with TRAILING missing FROM",
+        statement: "TRIM(trailing '')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM all args but string",
+        statement: "TRIM(TRAILING '' FROM)",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM two args no FROM",
+        statement: "TRIM(' ' '   1   ')",
+        assert: {
+            result: ParseError
+        },
+    },
+    parse::{
+        name: "TRIM spec and FROM no string",
+        statement: "TRIM(TRAILING FROM)",
+        assert: {
+            result: ParseError
+        },
+    },
+]
+
+extract::[
```

Comment:
For many of these, it is implementation-dependent as to whether it's a 'parse' error or a type/semantic error